include/yaclib/coroutine/detail/future_awaiter.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <yaclib/async/future.hpp>
+#include <yaclib/coroutine/detail/coroutine.hpp>
+#include <yaclib/util/detail/atomic_counter.hpp>
+#include <yaclib/util/type_traits.hpp>
+
+#include <iostream>
+
+namespace yaclib::detail {
+
+class FutureAwaiter final {
+ public:
+  FutureAwaiter() = delete;
+  FutureAwaiter(const FutureAwaiter&) = delete;
+  FutureAwaiter(FutureAwaiter&&) = delete;
+  FutureAwaiter& operator=(FutureAwaiter&&) = delete;
+  FutureAwaiter& operator=(const FutureAwaiter&) = delete;
+  ~FutureAwaiter() = default;
+
+  template <typename... V, typename... E>
+  explicit FutureAwaiter(Future<V, E>&... fs) : _await_core{1 + sizeof...(fs)} {
+    static_assert(sizeof...(fs) >= 1, "Number of futures must be at least one");
+    std::size_t wait_count = (... + (fs.GetCore()->SetWait(_await_core)));
+    _await_core.FetchSub(sizeof...(fs) - wait_count);
+  }
+
+  template <typename It, typename RangeIt>
+  explicit FutureAwaiter(It it, RangeIt begin, RangeIt end) : _await_core{static_cast<std::size_t>(1 + end - begin)} {
+    const auto count = end - begin;
+    std::size_t wait_count = 0;
+    for (; begin != end; ++begin, ++it) {
```

Comment:
NIT
I suggest while

Reply:
What does `NIT` mean? Why while is better?